### PR TITLE
[DCJ-324][risk=no] Add validation, tests, lower logging for Translate Resource

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-324

### Summary
* Add input validation to fail fast
* Lower logging - errors here do not need to go to Sentry
* Add tests for untested code paths and new validation

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/dsde/consent/ontology/resources/TranslateResource.java [hunk 1]
```diff
@@ -36,10 +34,13 @@ public TranslateResource(TextTranslationService translationService) {
...
   @Consumes(MediaType.APPLICATION_JSON)
   public Response translateSummary(String restriction) {
+    if (StringUtils.isBlank(restriction)) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
     try {
       return Response.ok().entity(translationService.translateDataUseSummary(restriction)).build();
     } catch (Exception e) {
-      logException("Error while translating", e);
+      logWarn("Error while translating restriction: " + e.getMessage());
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/dsde/consent/ontology/resources/TranslateResource.java:43]: Can we get a test for the exception?

File: src/main/java/org/broadinstitute/dsde/consent/ontology/resources/TranslateResource.java [hunk 2]
```diff
@@ -50,11 +51,14 @@ public Response translateSummary(String restriction) {
...
   public Response translate(@QueryParam("for") String forParam, String restriction) {
+    if (StringUtils.isBlank(forParam) || StringUtils.isBlank(restriction)) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
     try {
       TranslateFor translateFor = TranslateFor.find(forParam);
       return buildResponse(translateFor, restriction);
     } catch (Exception e) {
-      logException("Error while translating", e);
+      logWarn("Error while translating restriction: " + e.getMessage());
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/dsde/consent/ontology/resources/TranslateResource.java:61]: Ditto, can we get a test for the exception?

File: src/main/java/org/broadinstitute/dsde/consent/ontology/resources/TranslateResource.java [hunk 3]
```diff
@@ -75,7 +82,7 @@ public Response translateParagraph(final String jsonString) {
       return buildResponse(PARAGRAPH, paragraph);
     } catch (Exception e) {
       String message = "Server Error translating paragraph";
-      logException(message, e);
+      logWarn("Error while translating paragraph: " + e.getMessage());
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/dsde/consent/ontology/resources/TranslateResource.java:85]: Ditto, can we get a test for the exception?

**Review by fboulnois — APPROVED** [review]
Could we increase coverage by also testing the exceptions?

**Review by aarohinadkarni — APPROVED** [review]
looks good! 👍

**Pull request merged**